group-coordinator/src/test/java/org/apache/kafka/coordinator/group/assignor/ServerSideStickyRangeAssignorTest.java

```diff
@@ -0,0 +1,389 @@
+        Map<String, AssignmentMemberSpec> members = new HashMap<>();
+        // Consumer A
+        Map<Uuid, Set<Integer>> currentAssignmentForA = new HashMap<>();
+        currentAssignmentForA.put(topic1Uuid, new HashSet<>(Arrays.asList(0, 1, 2)));
+        currentAssignmentForA.put(topic2Uuid, new HashSet<>(Collections.singletonList(0)));
+        // Change subscriptions
+        List<Uuid> subscribedTopicsA = new ArrayList<>(Collections.singletonList(topic1Uuid));
+        members.computeIfAbsent(consumerA, k -> new AssignmentMemberSpec(Optional.empty(), Optional.empty(), subscribedTopicsA, currentAssignmentForA));
+        // Consumer B
+        Map<Uuid, Set<Integer>> currentAssignmentForB = new HashMap<>();
+        currentAssignmentForB.put(topic2Uuid, new HashSet<>(Collections.singletonList(1)));
+        // Change subscriptions
+        List<Uuid> subscribedTopicsB = new ArrayList<>(Arrays.asList(topic1Uuid, topic2Uuid, topic3Uuid));
+        members.computeIfAbsent(consumerB, k -> new AssignmentMemberSpec(Optional.empty(), Optional.empty(), subscribedTopicsB, currentAssignmentForB));
+        // Consumer C
+        Map<Uuid, Set<Integer>> currentAssignmentForC = new HashMap<>();
+        currentAssignmentForC.put(topic2Uuid, new HashSet<>(Collections.singletonList(2)));
+        currentAssignmentForC.put(topic3Uuid, new HashSet<>(Arrays.asList(0, 1)));
+        // Change subscriptions
+        List<Uuid> subscribedTopicsC = new ArrayList<>(Collections.singletonList(topic2Uuid));
+        members.put(consumerC, new AssignmentMemberSpec(Optional.empty(), Optional.empty(), subscribedTopicsC, new HashMap<>()));
+
+        AssignmentSpec assignmentSpec = new AssignmentSpec(members, topics);
+        GroupAssignment computedAssignment = assignor.assign(assignmentSpec);
+
+        Map<Uuid, Set<Set<Integer>>> expectedAssignment = new HashMap<>();
+        // Topic 1 Partitions Assignment
+        expectedAssignment.computeIfAbsent(topic1Uuid, k -> new HashSet<>()).add(new HashSet<>(Arrays.asList(0, 1)));
+        expectedAssignment.computeIfAbsent(topic1Uuid, k -> new HashSet<>()).add(new HashSet<>(Collections.singletonList(2)));
+        // Topic 2 Partitions Assignment
+        expectedAssignment.computeIfAbsent(topic2Uuid, k -> new HashSet<>()).add(new HashSet<>(Arrays.asList(0, 1)));
+        expectedAssignment.computeIfAbsent(topic2Uuid, k -> new HashSet<>()).add(new HashSet<>(Collections.singletonList(2)));
+        // Topic 3 Partitions Assignment
+        expectedAssignment.computeIfAbsent(topic3Uuid, k -> new HashSet<>()).add(new HashSet<>(Arrays.asList(0, 1)));
+
+        assertAssignment(expectedAssignment, computedAssignment);
+    }
+
+    // We have a set of sets with the partitions that should be distributed amongst the consumers, if it exists then remove it from the set.
+    private void assertAssignment(Map<Uuid, Set<Set<Integer>>> expectedAssignment, GroupAssignment computedGroupAssignment) {
```

Comment:
for expectedAssignment, can we add a layer for each memberId? this is blindly matching with any member's assignment which doesn't seem right.

then we won't have to remove from the expectedAssignment which is also not ideal

Reply:
No, this was done because we don't know which member gets which assignment. The order of members is not guaranteed at the time of assignment since they are stored in a hash map. So we're testing if the sets are correct and not testing the exact 1:1 mapping